backend/danswer/server/manage/search_settings.py

```diff
@@ -45,17 +60,28 @@ def set_new_search_settings(
     if search_settings_new.index_name:
         logger.warning("Index name was specified by request, this is not suggested")
 
-    # Validate cloud provider exists
+    # Validate cloud provider exists or create new LiteLLM provider
     if search_settings_new.provider_type is not None:
-        cloud_provider = get_embedding_provider_from_provider_type(
-            db_session, provider_type=search_settings_new.provider_type
-        )
-
-        if cloud_provider is None:
-            raise HTTPException(
-                status_code=status.HTTP_400_BAD_REQUEST,
-                detail=f"No embedding provider exists for cloud embedding type {search_settings_new.provider_type}",
+        if search_settings_new.provider_type == EmbeddingProvider.LITELLM:
+            # Create new LiteLLM provider if it doesn't exist
+            cloud_provider = get_embedding_provider_from_provider_type(
+                db_session, provider_type=EmbeddingProvider.LITELLM
             )
+            if cloud_provider is None:
+                # Logic to create new LiteLLM provider
```

Comment:
can we align the logic to be more the same between this and the other cloud providers (like just add a nullable URL field that only the LiteLLM proxy implementation would use)